File: app/src/main/java/ru/ifmo/md/exam1/PlaylistActivity.java
package ru.ifmo.md.exam1;

import android.app.LoaderManager;
import android.content.ContentUris;
import android.content.CursorLoader;
import android.content.Intent;
import android.content.Loader;
import android.database.Cursor;
import android.net.Uri;
import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import java.util.ArrayList;


public class PlaylistActivity extends ActionBarActivity implements LoaderManager.LoaderCallbacks<Cursor>{


    private ListView listView;
    private MyAdapter adapter;
    int plId;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_playlist);
        plId = getIntent().getIntExtra("plId",0);

        listView = (ListView) findViewById(R.id.play_list);
        adapter = new MyAdapter(new ArrayList<Track>());
        listView.setAdapter(adapter);
        listView.setOnItemClickListener(listListener);
        getLoaderManager().initLoader(1,null,this);

    }

    AdapterView.OnItemClickListener listListener = new AdapterView.OnItemClickListener() {
        @Override
        public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
            Intent intent = new Intent(PlaylistActivity.this,InfoActivity.class);
            int dbid = adapter.data.get(position).id;
            intent.putExtra("dbId",dbid);
            startActivity(intent);
        }
    };


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_edit, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
        return new CursorLoader(this, MyProvider.PHOTOS_CONTENT_URI,null,DBHelper.KEY_RAW_TYPE + " = " + 1,null,null);
    }

    @Override
    public void onLoadFinished(Loader<Cursor> loader, Cursor cursor) {
        adapter.data.clear();
        while (cursor.moveToNext()) {
            String name = cursor.getString(1);
            int id = cursor.getInt(0);
            String playlist = cursor.getString(7);

            if(playlist.contains(Integer.toString(plId))) {
                Track track = new Track();
                track.name = name;
                track.id = id;
                adapter.data.add(track);
            }
        }
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onLoaderReset(Loader<Cursor> loader) {

    }
}

File: app/src/main/scala/me/volhovm/android/exam2/PlaylistActivity.scala
package me.volhovm.android.exam2

import android.app.Activity
import android.app.LoaderManager.LoaderCallbacks
import android.content.{AsyncTaskLoader, Loader}
import android.os.Bundle
import android.view.{Menu, MenuItem}
import android.widget.{ArrayAdapter, ListView, Toast}

class PlaylistActivity extends Activity with LoaderCallbacks[PlayList] {
  private var mPlaylistName: String = null
  private var mPlaylist: PlayList = null
  private var mDatabaseHelper: DatabaseHelper = null
  private var mListView: ListView = null
  private var mListViewAdapter: ArrayAdapter[String] = null

  override def onCreate(savedInstanceState: Bundle): Unit = {
    super.onCreate(savedInstanceState)
    mPlaylistName = getIntent.getStringExtra("playlist_name")
    mDatabaseHelper = new DatabaseHelper(this)
    mListView = new ListView(this)
    setContentView(mListView)
    getLoaderManager.initLoader(0, null, this).forceLoad()
  }

  override def onCreateOptionsMenu(menu: Menu): Boolean = {
    getMenuInflater.inflate(R.menu.menu_auxiliary, menu)
    true
  }

  override def onOptionsItemSelected(item: MenuItem) = item.getItemId match {
    case R.id.action_settings => true
    case R.id.action_refresh =>
      Toast.makeText(this, "TEST", Toast.LENGTH_LONG).show()
      super.onOptionsItemSelected(item)
      true
  }
  override def onCreateLoader(p1: Int, p2: Bundle): Loader[PlayList] =
    new AsyncTaskLoader[PlayList](this) {
      override def loadInBackground(): PlayList = mDatabaseHelper.mWrapper.getPlaylist(mPlaylistName)
    }
  override def onLoaderReset(p1: Loader[PlayList]): Unit = ()
  override def onLoadFinished(p1: Loader[PlayList], p2: PlayList): Unit = {
    mPlaylist = p2
    mListViewAdapter = new ArrayAdapter[String](this, android.R.layout.simple_list_item_1,
      mPlaylist.songs.map((s: Song) => s.getArtist + " - " + s.getName).toArray)
    mListView.setAdapter(mListViewAdapter)
    mListViewAdapter.notifyDataSetChanged()
  }
}